lib/elixir/lib/macro.ex

```diff
@@ -2636,6 +2636,29 @@ defmodule Macro do
     end
   end
 
+  defp dbg_ast_to_debuggable({:case, _meta, [expr, [do: clauses]]} = ast) do
+    clause_index_var = unique_var(:clause_index, __MODULE__)
+    result_var = unique_var(:result, __MODULE__)
+    expr_var = unique_var(:expr, __MODULE__)
+
+    clauses_returning_index =
+      Enum.with_index(clauses, fn {:->, meta, [left, right]}, index ->
+        {:->, meta, [left, {right, index}]}
+      end)
+
+    quote do
+      unquote(expr_var) = unquote(expr)
+
+      {unquote(result_var), unquote(clause_index_var)} =
+        case unquote(expr_var) do
+          unquote(clauses_returning_index)
+        end
+
+      {:case, unquote(escape(ast)), unquote(expr_var), unquote(clause_index_var),
+       unquote(result_var)}
```

Comment:
Since we're not doing recursion or anything where you need to reuse the same variables, I _think_ you could not define the unique_vars here? 🤔 

```suggestion
      expr = unquote(expr)

      {result, clause_index} =
        case expr do
          unquote(clauses_returning_index)
        end

      {:case, unquote(escape(ast)), expr, clause_index, result}
```

Reply:
OMG you're right, how did I miss that 😅
Fixed in [6dc6bdf](https://github.com/elixir-lang/elixir/pull/12472/commits/6dc6bdfe244c6326dd86e926312158bf43609a24)

Thank you for all the suggestions ❤️